jerry-ext/include/jerryscript-ext/section.impl.h

```diff
@@ -0,0 +1,84 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRYX_SECTION_IMPL_H
+#define JERRYX_SECTION_IMPL_H
+
+/**
+ * Define the name of a section
+ *
+ * @param name the name of the section without quotes
+ */
+#ifdef __MACH__
+#define JERRYX_SECTION_NAME(name) "__DATA," #name
```

Comment:
You mean for systems that do not support linker sections?

An alternative for such systems is to use a constructor, for which we now have support in jerry-libc. However, constructors are also stored in a special linker section, so that too might not work either.

An uglier alternative if all else fails is to use code generation and parse the source code to extract `JERRYX_MODULE()` declarations.

Either way, we can define `JERRYX_MODULE()` in various different platform-specific ways.